src/openms_gui/source/VISUAL/ICONS/sequence_viz.html

```diff
@@ -0,0 +1,252 @@
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>Sequence Visualizer</title>
+    <script src="qrc:///qtwebchannel/qwebchannel.js"></script>
+    <script src="https://cdn.jsdelivr.net/npm/babel-polyfill" defer></script>
+    <!-- Web component polyfill (only loads what it needs) -->
+    <script src="https://cdn.jsdelivr.net/npm/@webcomponents/custom-elements@latest"
+            defer></script>
+    <!-- Required to polyfill modern browsers as code is ES5 for IE... -->
+    <script src="https://cdn.jsdelivr.net/npm/@webcomponents/custom-elements@latest/src/native-shim.js"
+            defer></script>
+
+    <!-- JQuery -->
+    <script src="https://code.jquery.com/jquery-3.6.0.min.js"></script>
+    <!-- d3 -->
+    <script src="https://cdn.jsdelivr.net/npm/d3@5.9.2" defer></script>
+
+    <!-- protvista components -->
+    <script src="https://cdn.jsdelivr.net/npm/protvista-manager@latest" defer></script>
+    <script src="https://cdn.jsdelivr.net/npm/protvista-utils@latest" defer></script>
+    <script src="https://cdn.jsdelivr.net/npm/protvista-zoomable@latest" defer></script>
+    <script src="https://cdn.jsdelivr.net/npm/protvista-track@latest" defer></script>
+    <script src="https://cdn.jsdelivr.net/npm/protvista-navigation@latest" defer></script>
+    <script src="https://cdn.jsdelivr.net/npm/protvista-tooltip@latest" defer></script>
+    <script src="https://cdn.jsdelivr.net/npm/protvista-sequence@latest" defer></script>
+    <script defer>
+        document.addEventListener("DOMContentLoaded", function () {
+            new QWebChannel(qt.webChannelTransport,
+            async function(channel){
+
+            let SequenceVisualizer = channel.objects.SequenceVisualizer;
+
+            function hashCode(str) {
+                let hash = 0;
+                for (let i = 0; i < str.length; i++) {
+                    hash = str.charCodeAt(i) + ((hash << 5) - hash);
+                }
+                return hash;
+            }
+
+            function pickColor(str) {
+                return `hsl(${hashCode(str) % 360}, 100%, 80%)`;
+            }
+
+            let acc_num = SequenceVisualizer.json_data_obj.accession_num;
+            let protein_seq = SequenceVisualizer.json_data_obj.protein_sequence_data;
+            let peptides_data_arr = SequenceVisualizer.json_data_obj.peptides_data;
+            let protlen = SequenceVisualizer.json_data_obj.protein_sequence_data.length;
+
+            let peptide_obj = [];
+            let mod_peptide_obj = [];
+            
+            if(!protein_seq.length){ 
+                async function fetchSeqDataJSON() {
+                  const response = await fetch('https://www.ebi.ac.uk/proteins/api/proteins/' + acc_num, {
+                        method:'GET',
+                        headers: {
+                            'Content-Type': 'application/json'
+                        }
+                  })
+                  if (!response.ok) {
```

Comment:
what if ebi website is down?

Then it throws an error. Ok.
But what does that mean for the interface? Is it a messagebox or does it close the app?

Reply:
I think it leaves the "browser" window half loaded without the sequence and throws an error in the "website log" (i.e. in the "Chrome tools" or equivalent view)